src/theory/arith/arith_rewriter.cpp

```diff
@@ -188,6 +188,91 @@ RewriteResponse ArithRewriter::postRewriteAtom(TNode atom)
   rewriter::addToSum(sum, left, negate);
   rewriter::addToSum(sum, right, !negate);
 
+  if (kind != Kind::EQUAL)
+  {
+    bool convertible = true;
+    // the (single) bv2nat term in the sum
+    Node bv2natTerm;
+    // whether the bv2nat term is positive in the sum
+    bool bv2natPol = false;
+    // the remaining sum (constant)
+    std::vector<Node> otherSum;
+    NodeManager* nm = NodeManager::currentNM();
+    for (std::pair<const Node, RealAlgebraicNumber>& m : sum)
+    {
+      if (m.second.isRational())
+      {
+        const Rational& r = m.second.toRational();
+        Kind mk = m.first.getKind();
+        if (mk == BITVECTOR_TO_NAT)
+        {
+          // We currently only eliminate sums involving exactly one
+          // (bv2nat x) monomial whose coefficient is +- 1, although more
+          // cases could be handled here.
```

Comment:
Perhaps it makes sense to add an issue pointing to this?

Reply:
I don't think this warrants an issue, since its not clear whether this is important.